### PR TITLE
[pigeon] Add macOS Obj-C support

### PR DESCRIPTION
**Pull request by stuartmorgan**
Makes Objective-C a supported language for macOS in Pigeon, and adds integration tests for it.

Also adjusts CI to only run the macOS version of the integration tests, as is already the case for Swift, since there's less complexity in running macOS integration tests than Obj-C integration tests, and the tests are identical. (As with Swift, we can easily revisit that if we ever find that we miss a bug due to this.)

Fixes https://github.com/flutter/flutter/issues/124724

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/pigeon/lib/objc_generator.dart
Comment (stuartmorgan) [inline packages/pigeon/lib/objc_generator.dart:1]: This is the actual fix; everything else is for testing :)

File: packages/pigeon/platform_tests/alternate_language_test_plugin/macos/Classes/AlternateLanguageTestPlugin.h
Comment (stuartmorgan) [inline packages/pigeon/platform_tests/alternate_language_test_plugin/macos/Classes/AlternateLanguageTestPlugin.h:1]: This and the .m file are direct copies of the iOS version, with just the `import` changed.

File: packages/pigeon/platform_tests/alternate_language_test_plugin/example/macos/RunnerTests/RunnerTests.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import FlutterMacOS
```
Comment (tarrinneal) [inline packages/pigeon/platform_tests/alternate_language_test_plugin/example/macos/RunnerTests/RunnerTests.swift:1]: Needs a license header I think

**Review by tarrinneal — APPROVED** [review]

File: packages/pigeon/platform_tests/alternate_language_test_plugin/example/macos/RunnerTests/RunnerTests.swift (hunk 1 above)
Comment (stuartmorgan) [reply packages/pigeon/platform_tests/alternate_language_test_plugin/example/macos/RunnerTests/RunnerTests.swift:5]: I checked locally, but had some noise from generated files and missed it :(

**Pull request merged**